### PR TITLE
feat(LERN-76): added access-policy adr

### PR DESCRIPTION
**Pull request by A1eeee**

### CODE REVIEW

File: doc/adr/0003-access-policy.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# 3. Access Policy
+
+Date: 2024-05-21
+
+## Status
+
+In Progress
```
Comment (deepspace1000) [inline doc/adr/0003-access-policy.md:7]:
```suggestion
Proposed
```
🙃 but doesn't really matter

File: doc/adr/0003-access-policy.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+## Context
+
+The application necessitates the creation of distinct user types, each possessing unique access privileges. The challenge we face is the development of a role-based access control system that is both secure and scalable. The roles we have identified thus far include Student, Teacher, and Admin.
+
+## Decision
+
+We have decided to implement a role-based access control system across our entire backend application. This approach will enable us to specify which roles have access to which resources, thereby securing and protecting specific endpoints.
+
+We will create separate packages for each user role. This strategy will facilitate the easy implementation of new features for specific roles and will ensure a well-structured backend application.
```
Comment (deepspace1000) [inline doc/adr/0003-access-policy.md:17]: I would add something like. 
`We use the same root url through out the package e.g. for the admin package all resource endpoint start with /admin. Those root endpoints can only be called by a use with corresponding role.`

File: doc/adr/0003-access-policy.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+We will be working with the following roles and packages:
+- Admin
+- Teacher
+- Student
+
+## Consequences
+
+This decision will enable us to implement a secure and scalable access control system for our application. 
+The most significant challenge will be the implementation of the access control, which will require us to work closely with our security handler (SpringBoot Security). 
+If we succeed in implementing the access policy, we will eliminate the need for PreAuthorization for the user role in each resource.
```
Comment (deepspace1000) [inline doc/adr/0003-access-policy.md:28]: you can remove the last point. 

```suggestion
```

File: doc/adr/0003-access-policy.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+We will create separate packages for each user role. This strategy will facilitate the easy implementation of new features for specific roles and will ensure a well-structured backend application.
+
+We will be working with the following roles and packages:
+- Admin
+- Teacher
+- Student
+
+## Consequences
+
```
Comment (deepspace1000) [inline doc/adr/0003-access-policy.md:25]: Probably add something like:
`we can control access through out a root url with just the role`
not my best text, but i think you get what i mean

**Review by deepspace1000 — COMMENTED** [review]
Nice job thats what i wanted 🥳. just some little things about the contents

**Comment by lost-pixel-signals** [comment]
| Shot Type       | Count                | Status   |
| -------------   | -------------------- | -------- |
| Added           | 0       | 🟡       |
| Deleted         | 0     | ❌       |
| Regression      | 0  | ❗       |

**Review by deepspace1000 — APPROVED** [review]
Nice

**Pull request merged**